include/extractor/guidance/intersection.hpp

```diff
@@ -108,11 +115,60 @@ struct ConnectedRoad final : IntersectionViewData
 };
 
 // small helper function to print the content of a connected road
+std::string toString(const IntersectionShapeData &shape);
+std::string toString(const IntersectionViewData &view);
 std::string toString(const ConnectedRoad &road);
 
 // Intersections are sorted roads: [0] being the UTurn road, then from sharp right to sharp left.
+// common operations shared amongst all intersection types
+template <typename Self> struct EnableShapeOps
+{
+    // same as closest turn, but for bearings
+    auto findClosestBearing(double bearing) const
+    {
+        auto comp = makeCompareShapeDataByBearing(bearing);
+        return std::min_element(self()->begin(), self()->end(), comp);
+    }
+
+    // search a given eid in the intersection
+    auto findEid(const EdgeID eid) const
+    {
+        return std::find_if(
+            self()->begin(), self()->end(), [eid](const auto &road) { return road.eid == eid; });
+    }
+
+    // find the maximum value based on a conversion operator
+    template <typename UnaryPredicate> auto findMaximum(const UnaryPredicate converter) const
+    {
+        BOOST_ASSERT(!self()->empty());
```

Comment:
Can we put a single assertion at locations where we generate `Intersection`s / `IntersectionView`s instead of doing it for every function call then? Roads should be immutable or do we ever remove / add roads?